internal/renderer/service_util.go

```diff
@@ -341,6 +343,51 @@ func createLbService4Gateway(c *RenderContext, gw *gwapiv1a2.Gateway) *corev1.Se
 		})
 	}
 
+	// Set health check port
```

Comment:
Can we refactor this into a separate function? The function should receive the gwconf, the gw and the service and then (1) try to match the hcPort and return immediately if it couldn't, (2) try to match the hcProto and return if it couldn't (I'm not sure about whether we should default to TCP is no annotation was found, I'd rather avoid smart defaults), and (3) if all goes well it'd add the service port and log.